packages/task/src/browser/quick-open-task.ts

```diff
@@ -47,11 +59,21 @@ export class QuickOpenTask implements QuickOpenModel, QuickOpenHandler {
         const configuredTasks = this.taskService.getConfiguredTasks();
         const providedTasks = await this.taskService.getProvidedTasks();
 
+        const filteredProvidedTasks: TaskConfiguration[] = [];
+        providedTasks.forEach(provided => {
+            if (!configuredTasks.some(configured => configured.label === provided.label)) {
+                filteredProvidedTasks.push(provided);
+            }
+        });
+
         this.items = [];
         this.items.push(
             ...configuredTasks.map((t, ind) => new TaskRunQuickOpenItem(t, this.taskService, true, ind === 0 ? 'configured' : undefined)),
-            ...providedTasks.map((t, ind) => new TaskRunQuickOpenItem(t, this.taskService, false, ind === 0 ? 'provided' : undefined))
+            ...filteredProvidedTasks.map((t, ind) => new TaskRunQuickOpenItem(t, this.taskService, false, ind === 0 ? 'provided' : undefined))
         );
+
+        this.actionProvider = this.items.length ? this.taskActionProvider : this.noActionProvider;
```

Comment:
Do we need `NoActionProvider`? How about assigning `undefined` instead?

Reply:
I used `NoActionProvider` to avoid potential NPE, but I reviewed code - we have [check](https://github.com/theia-ide/theia/pull/4714/files#diff-fd1f520aac49f5b20bec0b65f749c89eR230), so I removed this entry.